src/views/Home.jsx

```diff
@@ -1,11 +1,29 @@
 import './Home.css';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import { useNavigate } from 'react-router-dom';
+import { useEffect } from 'react';
+
+export function Home({ listToken, setListToken }) {
+	const navigate = useNavigate();
+
+	const createToken = async () => {
+		const token = await generateToken();
+
+		setListToken(token);
+	};
+
+	useEffect(() => {
+		if (listToken) {
+			navigate('/list');
+		}
+	}, [listToken, navigate]);
```

Comment:
Do we need the useEffect in the home component? It's already present on the App component... That feels like a duplicate (unless I'm missing something)

Reply:
I've taken navigate out of the dependency array for this useEffect hook, I'm not sure if navigate is causing infinite re-renders, I think it's plausible, but nevertheless it's unnecessary to include it in the dependencies array as we're only watching for changes in listToken (no token => token created)

Reply:
I learned something new about useEffect today :) (being able to add a dependency. And yes, I don't know that adding or removing navigate from the dependency makes a difference, but since we don't need to listen to it, I think it's prudent to remove it like you suggested.  I think the loop was occurring from the App.jsx useEffect.